posthog/middleware.py

```diff
@@ -270,3 +276,80 @@ def __call__(self, request: HttpRequest):
                 reset_query_tags()
         response: HttpResponse = self.get_response(request)
         return response
+
+
+class CaptureMiddleware:
+    """
+    Middleware to serve up capture responses. We specifically want to avoid
+    doing any unnecessary work in these endpoints as they are hit very
+    frequently, and we want to provide the best availability possible, which
+    translates to keeping dependencies to a minimum.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+        middlewares: List[Any] = []
+        # based on how we're using these middlewares, only middlewares that
+        # have a process_request and process_response attribute can be valid here.
+        # Or, middlewares that inherit from `middleware.util.deprecation.MiddlewareMixin` which
+        # reconciles the old style middleware with the new style middleware.
+        for middleware_class in (
+            PrometheusBeforeMiddleware,
+            SecurityMiddleware,
+            CorsMiddleware,
+            PrometheusAfterMiddleware,
+        ):
+            try:
+                # Some middlewares raise MiddlewareNotUsed if they are not
+                # needed. In this case we want to avoid the default middlewares
+                # being used.
+                middlewares.append(middleware_class(get_response=None))
+            except MiddlewareNotUsed:
+                pass
+
+        self.CAPTURE_MIDDLEWARE = middlewares
+
+        if STATSD_HOST is not None:
+            self.CAPTURE_MIDDLEWARE.insert(0, StatsdMiddleware())
+            self.CAPTURE_MIDDLEWARE.append(StatsdMiddlewareTimer())
+
+    def __call__(self, request: HttpRequest):
+
+        if request.path in (
+            "/e",
+            "/e/",
+            "/s",
+            "/s/",
+            "/track",
+            "/track/",
+            "/capture",
+            "/capture/",
+            "/batch",
+            "/batch/",
+            "/engage/",
+            "/engage",
+        ):
+            try:
+                # :KLUDGE: Manually tag ClickHouse queries as CHMiddleware is skipped
```

Comment:
Any reason not to add the CHQuery middleware to the list of middlewares above?

Reply:
It's the new style, I'd need to rewrite it in the old-style, or adjust to accomodate new style middlewares somehow.

Reply:
Ah right, no need in this PR